packages/ERTP/src/legacy-payment-helpers.js

```diff
@@ -0,0 +1,104 @@
+import { mustMatch } from '@agoric/store';
+import { E } from '@endo/far';
+import { AmountMath } from './amountMath.js';
+
+const { Fail } = assert;
+
+// This file contains safer alternatives to the similarly named helpers on
+// issuer. These are parameterized by a purse. Any payments created by these
+// helper functions are in the recovery set of that purse until otherwise
+// used up.
+//
+// One helper is less safe in one way: `combine` is not failure atomic. If
+// it fails, some of the input payments may have been used up. However, even
+// in that case, no assets should be lost. The assets from the used up payments
+// will be in the argument purse.
+
+/**
+ * @template {AssetKind} K
+ * @param {ERef<Purse<K>>} purse
+ * @param {ERef<Payment<K>>} srcPaymentP
+ * @param {Pattern} [optAmountShape]
+ * @returns {Promise<Payment<K>>}
+ */
+export const claim = async (purse, srcPaymentP, optAmountShape = undefined) => {
+  const srcPayment = await srcPaymentP;
+  const amount = await E(purse).deposit(srcPayment, optAmountShape);
+  return E(purse).withdraw(amount);
```

Comment:
Would this be better?
```suggestion
  return E.when(
    E(purse).deposit(srcPayment, optAmountShape),
    amount => E(purse).withdraw(amount),
  );
```

Reply:
Done